internal/pkg/fieldtags/analyzer.go

```diff
@@ -82,17 +82,16 @@ func run(pass *analysis.Pass) (interface{}, error) {
 		}
 		for _, f := range (*s).Fields.List {
 			if patterns.isSource(f) && len(f.Names) > 0 {
-				tf := TaggedField{
-					pkg:   pass.Pkg.Name(),
-					typ:   t.Name.Name,
-					field: f.Names[0].Name,
-				}
-				taggedFields = append(taggedFields, tf)
-				pass.Reportf(f.Pos(), "tagged field: %s", tf)
+				pass.ExportObjectFact(pass.TypesInfo.ObjectOf(f.Names[0]), &isTaggedField{})
 			}
 		}
 	})
-	return taggedFields, nil
+
+	taggedFields := TaggedFields{}
+	for _, of := range pass.AllObjectFacts() {
```

Comment:
I believe this will also return the objects for any imported dependency, which do not seem useful to downstream analyzers.  See about about `Fact`s in general.